hypercall.h

```diff
@@ -10,11 +10,12 @@ int strnlen_n(char* s, int max_len){
 
 // #if defined(__x86_64__)
 #if defined(__x86_64__)
-static inline int hc(char *s) {
+static inline int hc(char **s,int len) {
     uint64_t eax = MAGIC_VAL;
     uint64_t ret = MAGIC_VAL;
-    uint64_t action = (uint32_t)strnlen_n(s, 0x1000);
-
+    for(int i = 0; i< len; i++){
```

Comment:
I don't love this design - why not just call strnlen once? That should iterate through every byte in your input and page it it. My algorithms knowledge is rusty but this seems to be O(n^2) and I don't think it needs to be

Reply:
Oh, looking at this again, s is a `char**` not a `char*` so the loop does make sense!